sklearn/preprocessing/_discretization.py

```diff
@@ -285,9 +289,7 @@ def inverse_transform(self, Xt):
         check_is_fitted(self, ["bin_edges_"])
 
         if self.encode != 'ordinal':
-            raise ValueError("inverse_transform only supports "
-                             "'encode = ordinal'. Got encode={!r} instead."
-                             .format(self.encode))
+            Xt = self.ohe_encoder_.inverse_transform(Xt)
```

Comment:
For inverse transform, we need to use check_array to do input validation and then check number of features. So you'll have something like
```
Xinv = check_array...
if self.encode != 'ordinal':
    if ...
        raise ValueError("Incorrect number of features....)
    Xinv = self.ohe_encoder_.inverse_transform(Xinv)
else:
   ...
```
Also, maybe we should make ohe_encoder_ a private attribute, otherwise we need to document it.

Reply:
Maybe a better way is to rely on the input validation provided by OneHotEncoder.inverse_transform, so you'll have something like
```
if self.encode != 'ordinal':
    self.ohe_encoder_.inverse_transform
else:
    existing input validation
```

Reply:
Yes OneHotEncoder already run input validation, doesn't the output of the OHE inverse_transform need to be validated as well? I mean is there some case in which the input can pass the validation for a OneHotEncoder and not for a KBinsDiscretizer? As it is now both validation are runt.

I was thinking about the private variable as well, I have a doubt though, that could be a very stupid question but how can I run all the tests if the ohe_encoder is private? In particular 
```
 if encode != 'ordinal':
         Xt_tmp = kbd.ohe_encoder_.inverse_transform(X2t)
     else:
         Xt_tmp = X2t
     assert_array_equal(Xt_tmp.max(axis=0) + 1, kbd.n_bins_)
```

Should I write a different test?